File: Aplicaciones Distribuidas/PARCIAL1/05. HILOS_NETBEANS_CHINLLE_CHUQUIMARCA_COBACANGO/TALLER_HILOS_PRACTICA3/Aplicativo/178_TALLER_HILOS_GRUPO05/src/ec/edu/monster/controlador/TransferenciasControlador.java
package ec.edu.monster.controlador;

import ec.edu.monster.modelo.BancoModelo;

public class TransferenciasControlador implements Runnable {
    private BancoModelo _banco;
    private BancoControlador _controller;
    private int _cuentaOrigen;
    private double _valorMaximo;

    public TransferenciasControlador(BancoModelo banco, BancoControlador controller, int cuentaOrigen, double valorMaximo) {
        this._banco = banco;
        this._controller = controller;
        this._cuentaOrigen = cuentaOrigen;
        this._valorMaximo = valorMaximo;
    }

    @Override
    public void run() {
        while (true) {
            int cuentaDestino = (int) (100 * Math.random());
            double cantidad = this._valorMaximo * Math.random();
            try {
                this._controller.transferirEntreCuentas(
                    this._banco,
                    this._cuentaOrigen, 
                    cuentaDestino,
                    cantidad
                );
            } catch (InterruptedException e1) {
                e1.printStackTrace();
            }

            try {
                Thread.sleep((int) (Math.random() * 10));
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

File: Aplicaciones Distribuidas/PARCIAL2/HILOS_DOTNET_GRUPO05/TALLER_HILOS_PRACTICA3/APLICATIVO/174_TALLER_HILOS_GRUPO05/Controlador/TransferenciasControlador.cs
using _174_TALLER_HILOS_GRUPO05.Modelo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace _174_TALLER_HILOS_GRUPO05.Controlador
{
    public class TransferenciasControlador
    {
        private BancoModelo _banco;
        private BancoControlador _controller;
        private int _cuentaOrigen;
        private double _valorMaximo;

        public TransferenciasControlador(BancoModelo banco, BancoControlador controller, int cuentaOrigen, double valorMaximo)
        {
            this._banco = banco;
            this._controller = controller;
            this._cuentaOrigen = cuentaOrigen;
            this._valorMaximo = valorMaximo;
        }

        public void run()
        {
            Random random = new Random();
            while (true)
            {
                int cuentaDestino = (int)(random.Next(100));
                double cantidad = random.NextDouble() * this._valorMaximo;
                this._controller.transferirEntreCuentas(this._banco, this._cuentaOrigen, cuentaDestino, cantidad);

                try
                {
                    Thread.Sleep(random.Next(10));
                }
                catch (Exception e)
                {
                    Console.WriteLine(e);
                }
            }
        }

    }
}
